skin/frontend/base/default/js/bold/checkout_payment_booster/expresspay.js

```diff
@@ -0,0 +1,730 @@
+     * @throws Error
+     */
+    const updateMagentoShippingMethod = async shippingOptions => {
+        let selectedShippingMethod;
+        let shippingMethodFormData;
+        let updateShippingMethodResponse;
+        let updateShippingMethodResult;
+
+        const shippingMethods = parseShippingMethodsFromHtml();
+
+        if (shippingMethods.length === 0) {
+            return;
+        }
+
+        if (shippingOptions === null) {
+            selectedShippingMethod = shippingMethods[0];
+        } else {
+            selectedShippingMethod = shippingMethods.find(shippingMethod => shippingMethod.code === shippingOptions.id);
+        }
+
+        if (selectedShippingMethod === undefined) {
+            return;
+        }
+
+        shippingMethodFormData = new FormData();
+
+        shippingMethodFormData.append('form_key', config.formKey);
+        shippingMethodFormData.append('shipping_method', selectedShippingMethod.code);
+
+        try {
+            updateShippingMethodResponse = await fetch(
+                config.saveShippingMethodUrl,
+                {
+                    method: 'POST',
+                    body: shippingMethodFormData
+                }
+            );
+        } catch (error) {
+            console.error(`Could not update Magento shipping method for Express Pay order.`, error);
+
+            throw error;
+        }
+
+        try {
+            updateShippingMethodResult = await updateShippingMethodResponse.json();
+        } catch (error) {
+            updateShippingMethodResult = {};
+        }
+
+        if (updateShippingMethodResult.hasOwnProperty('error') && updateShippingMethodResult.error) {
+            console.error(
+                `Could not update Magento shipping method for Express Pay order.`,
+                updateShippingMethodResult.message
+            );
+
+            throw new Error(updateShippingMethodResult.message);
+        }
+    };
+
+    /**
+     * @param {String} orderId
+     * @returns {Promise<void>}
+     * @throws Error
+     */
+    const placeMagentoOrder = async orderId => {
+        let placeOrderResponse;
+        let placeOrderResult;
+
+        const orderData = {
+            form_key: config.formKey,
+            payment: {
+                method: 'bold',
+                additional_data: {
+                    order_id: orderId
+                }
+            }
+        };
+
+        try {
+            placeOrderResponse = await fetch(
+                config.saveOrderUrl,
+                {
+                    method: 'POST',
+                    body: convertObjectToFormData(orderData)
+                }
+            );
+        } catch (error) {
+            console.error('Could not place Express Pay order in Magento.', error);
+
+            throw error;
+        }
+
+        /*if (placeOrderResponse.redirected) {
+            window.location.href = encodeURI(placeOrderResponse.url);
+
+            return;
+        }*/
```

Comment:
Do we want to keep this comment or should it be removed?

Reply:
That logic came from the core checkout functionality. I experimented with it and decided that it probably is not a good idea to redirect back to the cart page if an error occurs, but I left it in so that we have it if necessary in the future.